### PR TITLE
Adds Onboarding About You Screen

### PR DESCRIPTION
**Pull request by redreceipt**
## DESCRIPTION

Screen Shot 2019-03-18 at 5 12 00 PM

### What does this PR do, or why is it needed?

Adds the onboarding screen that asks for gender and birthday.

### What design trade-offs/decisions were made?

Using a `Radio` instead of the button group from the design. Using `DateInput` instead of the text field in from the design.

The date picker doesn't update, probably a change that needs to happen on the `ui-kit` side of things.

There is also a warning coming from the `DateInput`, also probably needs to be fixed in `ui-kit`

### How do I test this PR?

`yarn storybook`

### What automated tests did you write?

tested all props

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] Closes [#551]
- [ ] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by redreceipt** [comment]
@conrad-vanl I could use a little guidance on using `DateInput` and `RadioButton` if that is the best solution for putting this screen together.

Screen Shot 2019-03-14 at 10 51 18 AM

**Comment by redreceipt** [comment]
@conrad-vanl getting a new prop type warning trying to use `DateInput`, any ideas?

Screen Shot 2019-03-17 at 4 02 10 PM

**Comment by redreceipt** [comment]
After doing some digging, our `Radio` produces what you would expect an HTML radio select to look like so not exactly what we're going for. I'm going to look around for another option...

**Comment by redreceipt** [comment]
getting there but I'm just using a custom button group, is this the right route?

Screen Shot 2019-03-17 at 5 57 30 PM

cc @burkeshartsis @conrad-vanl

**Comment by redreceipt** [comment]
In thinking about this, I could make a `ButtonGroup` component that we could reuse in `ui-kit`. It would extend `Button` and simply take one prop, an array of strings.

**Comment by burkeshartsis** [comment]
Probably should be it's own component even if it's just extending `Button`. `Button` already has a ton of props.

**Comment by redreceipt** [comment]
@burkeshartsis I don't see a way I can make this `ButtonGroup` stateless, just wanna make sure I'm on the right path before starting on building this out.

**Comment by burkeshartsis** [comment]
I mentioned before but you are welcome to just use the radio buttons we already have. @conrad-vanl and I discussed it and for mvp it's find to just roll with the components we already have. The button group design isn't used anywhere else in the app and is actually an ios convention if I remember correctly. Which is to say, while it's not hard to understand it's only really familiar to ios users (I think).

**Comment by redreceipt** [comment]
so change the design to something like this?

() Male
() Female

**Comment by burkeshartsis** [comment]
Sure or ~float~ align them next to each other 

( ) Male     ( ) Female

**Comment by redreceipt** [comment]
@burkeshartsis I'm having trouble importing `Radio.Button` what's the correct syntax here?

Screen Shot 2019-03-18 at 9 45 23 AM

**Comment by redreceipt** [comment]
blocked by #597

File: packages/apolloschurchapp/src/ui/Onboarding/slides/AboutYou/index.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+const StyledRadio = styled(({ theme }) => ({
+  marginBottom: theme.sizing.baseUnit,
+  flexDirection: 'row',
+}))(Radio);
+
+const RadioLabel = styled(({ theme }) => ({
+  marginLeft: theme.sizing.baseUnit * 0.5,
+}))(H5);
+
+const genders = ['Male', 'Female'];
```
Comment (burkeshartsis) [inline packages/apolloschurchapp/src/ui/Onboarding/slides/AboutYou/index.js]: I'm not sure these need to be hardcoded in this way. A good component shouldn't make these kinds of assumptions but rather expose an api and render some sane defaults to keep the component as data driven as possible. I suggest moving this to a prop (`availableGenders`?) and setting this value as the default.

File: packages/apolloschurchapp/src/ui/Onboarding/slides/AboutYou/index.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+AboutYou.propTypes = {
+  /* The `Swiper` component used in `<Onboading>` looks for and hijacks the title prop of it's
+   * children. Thus we have to use more unique name.
+   */
+  slideTitle: PropTypes.string,
+  description: PropTypes.string,
+  imgSrc: PropTypes.string,
+  birthday: PropTypes.instanceOf(Date),
+  gender: PropTypes.oneOf(genders),
```
[the remark this answers is not in the document]
Comment (redreceipt) [reply packages/apolloschurchapp/src/ui/Onboarding/slides/AboutYou/index.js]: I feel like it shouldn't know about the user. `gender` seems descriptive enough given the component has no context of anything other than the gender field in the form.

[the remark this answers is not in the document]
Comment (redreceipt) [reply packages/apolloschurchapp/src/ui/Onboarding/slides/AboutYou/index.js]: nevermind, I changed my mind given passing in `genderList` too

File: packages/apolloschurchapp/src/ui/Onboarding/slides/AboutYou/index.js (hunk 1 above)
Comment (redreceipt) [reply packages/apolloschurchapp/src/ui/Onboarding/slides/AboutYou/index.js]: @burkeshartsis ok so I'm tracking here but how do I set up prop types for `userGender`? Really should be `oneOf(genderList)` which is another passed prop but you can't do that...

**Comment by redreceipt** [comment]
@burkeshartsis we could do that but now we're not testing the common use case which is a new person who doesn't have a birthday set already

**Comment by redreceipt** [comment]
@burkeshartsis would love to glean everything you learned about dates to make those tests work right :)

**Pull request merged**